lib/AST/NameLookup.cpp

```diff
@@ -1467,20 +1468,28 @@ DirectLookupRequest::evaluate(Evaluator &evaluator,
       Table.updateLookupTable(decl);
     }
   } else if (!Table.isLazilyComplete(name.getBaseName())) {
-    // The lookup table believes it doesn't have a complete accounting of this
-    // name - either because we're never seen it before, or another extension
-    // was registered since the last time we searched. Ask the loaders to give
-    // us a hand.
-    DeclBaseName baseName(name.getBaseName());
-    populateLookupTableEntryFromLazyIDCLoader(ctx, Table, baseName, decl);
-
-    if (!disableAdditionalExtensionLoading) {
-      populateLookupTableEntryFromExtensions(ctx, Table, baseName, decl);
+    if (decl->getClangDecl() &&
```

Comment:
In the future, I imagine we'll have a switch here, or a lot of if statements, that dispatch out to not just importing namespaces but also records, etc.